src/checkmatelib/client.py

```diff
@@ -0,0 +1,51 @@
+"""A client for the Checkmate URL testing service."""
+
+import requests
+from future.utils import raise_from  # Python 2.7 compatibility
+from requests.exceptions import ConnectionError as ConnectionError_
+from requests.exceptions import HTTPError, Timeout
+
+from checkmatelib._response import BlockResponse
+from checkmatelib.exceptions import CheckmateException
+
+# pylint: disable=too-few-public-methods
+
+
+class CheckmateClient:
+    """A client for the Checkmate URL testing service."""
+
+    def __init__(self, host):
+        """Initialise a client for contacting the Checkmate service.
+
+        :param host: The host including scheme, for the Checkmate service
+        """
+        self._host = host.rstrip("/")
+
+    def check_url(self, url):
+        """Check a URL for reasons to block.
+
+        :param url: URL to check
+        :raises CheckmateException: With any issue with the Checkmate service
+        :return: None if the URL is fine or a `CheckmateResponse` if there are
+           reasons to block the URL.
+        """
+        try:
+            response = requests.get(
+                self._host + "/api/check", params={"url": url}, timeout=1
+            )
+        except (ConnectionError_, Timeout) as err:
+            raise_from(CheckmateException("Cannot connect to service"), err)
+
+        try:
+            response.raise_for_status()
+        except HTTPError as err:
+            raise_from(CheckmateException("Unexpected response from service"), err)
```

Comment:
I think you might want to consider copying [how LMS does it](https://github.com/hypothesis/lms/blob/4afea105f545c2dc9620e94045f9fc00dae78dec/lms/services/canvas_api/_basic.py#L85-L89). I don't remember the details, but I remember that I took quite a few shots at dealing with `requests` and all the different ways it can raise, it was surprisingly difficult to catch them all, and finally came to that solution. Catching `RequestException` (the base exception for all requests exceptions) makes sure you get them all, and just allowing both the `requests.get()` and the `response.raise_for_status()` to be inside the same `try` reduced duplication

Reply:
There's something similar in Via 3 as well, which goes through quite a few permutations when checking URLs.

I wasn't sure what our stance on multiple things in a try is.

Reply:
Via 3 slaps a decorator around the whole thing and then differentiates a few different classes for reporting:

 * https://github.com/hypothesis/via3/blob/master/via/get_url/details.py
 * 
```
from requests import exceptions

REQUESTS_BAD_URL = (
    exceptions.MissingSchema,
    exceptions.InvalidSchema,
    exceptions.InvalidURL,
    exceptions.URLRequired,
)
REQUESTS_UPSTREAM_SERVICE = (
    exceptions.ConnectionError,
    exceptions.Timeout,
    exceptions.TooManyRedirects,
    exceptions.SSLError,
)
```

I think a simiar approach might be nice.

Reply:
I've pretty much ripped off Via 3